### PR TITLE
Proposal for themes implementation

### PR DESCRIPTION
**Pull request by evemontalvao**
**CHANGELOG** :memo:

* Thinking about how to implement the new identity by considering the following points:

  - Easily rollback to the old identity.
  - The possibility to test new themes in the future.

This PR is a possibility to build multiple themes on Gaiden. Each theme is a directory in `./src/scss`and the build is made by running `npm run build:ENV` to build all themes or `npm run build:ENV THEME_NAME` to build a specific theme. The generated file takes the name of the theme, example: `gaiden.yellow.min.css`.
Since multiple projects use Gaiden, we don't want to affect them or to create the need to updated all of them, so the current theme is now called "default" and it's build generates a file without theme name, just as it is today (`gaiden.min.css`).

**The number of files changed is this big because I had to change the entire structure of css files. But you can focus only in `bin/build/scss/ENV/css.js`** 😄 
I also had to move `bin` scripts to `scripts`, the reason is this error:
Screen Shot 2019-07-30 at 14 38 48

### CODE REVIEW

**Comment by iondrimba** [comment]
I like the idea of having multiple themes isolated from each other 👍

File: bin/build/dev/css.js [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
-    --source-map true -r \
-    --output-style expanded \
-    --sourceComments true \
-    -o ${outputDir}`,
-  postcss: `--use autoprefixer ${minifiedFile} -o ${minifiedFile}`,
-  sasslint: `-c .sass-lint.yml  -v -q`
-}
 
-const sassLintExec = shelljs.exec(`sass-lint ${options.sasslint}`);
+const execBuild = (theme) => {
```
Comment (williamcosta) [inline bin/build/dev/css.js]: Nice 👏

**Comment by williamcosta** [comment]
Another point, how would be our documentation in this case, we could change the theme?

I know we have a lagged documentation with JSDocs and probably with the development of Design System we will change this documentation, but just wanted to know if we think of any solution for the moment, or we would not have it from the current documentation.

**Comment by evemontalvao** [comment]
> Another point, how would be our documentation in this case, we could change the theme?
> 
> I know we have a lagged documentation with JSDocs and probably with the development of Design System we will change this documentation, but just wanted to know if we think of any solution for the moment, or we would not have it from the current documentation.

@williamcosta at this moment I don't have any solution for documentation. I guess I have to understand how JSDocs works first

**Comment by evemontalvao** [comment]
> @evemontalvao did you find it necessary to duplicate all files into the `yellow` folder?
> 
> Maybe we can have folders that are shared between themes where we wouldn't need to duplicate code, an example would be the `tools` folder where we have mixins, what do you think?

@williamcosta, @iondrimba also suggested this approach. I just moved `tools` and `generic` directories to a new folder called `shared` 😄

**Comment by eduardojmatos** [comment]
Nice!!! 👏 👏 👏

**Comment by eduardojmatos** [comment]
My suggestion about JSDocs it's not to complain about how to use now that and make studies about another tool to documentation our styleguide in other time :)

**Pull request merged**